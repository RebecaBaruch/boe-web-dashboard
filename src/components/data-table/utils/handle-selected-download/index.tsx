import { pdf } from '@react-pdf/renderer';
import {
  PDFDocumentGenerator,
  PDFDocumentProps,
} from '../../../../../src/utils/pdf-document-generator';

const downloadPDF = async (
  record: PDFDocumentProps,
  fileName: string = 'diagnosis_report.pdf',
) => {
  const pdfDocument = PDFDocumentGenerator(record) as React.ReactElement;

  const blob = await pdf(pdfDocument).toBlob();

  const link = document.createElement('a');
  link.href = URL.createObjectURL(blob);
  link.download = fileName;
  link.click();
};

export default downloadPDF;
