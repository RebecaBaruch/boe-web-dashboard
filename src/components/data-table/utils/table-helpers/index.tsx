import React from 'react';
import { Checkbox } from '@telefonica/mistica';
import { ActionButton, ActionModal, TableCell } from '../../styled';
import { MoreVertical } from 'react-feather';
import ActionButtons from 'components/data-table/action-buttons/action-buttons.component';

import {
  TableAccountsData,
  TableDiagnosisData,
  TableAnimalsData,
  Column,
} from '../../types';

interface SelectableCellProps {
  isSelected: boolean;
  onSelect: () => void;
}

interface ActionCellProps {
  onActionClick: () => void;
}

interface RenderActionCellProps {
  action?: 'download' | 'delete';
  selectMode?: boolean;
  isSelected: boolean;
  toggleSelectRow?: (id: string) => void;
  item: TableAccountsData | TableDiagnosisData | TableAnimalsData;
  rowIndex: number;
  onAction: (rowIndex: number) => void;
}

interface RenderActionModalProps {
  activeRowIndex: number | null;
  rowIndex: number;
  action?: 'download' | 'delete';
  modalRef: React.RefObject<HTMLDivElement>;
  data: TableAccountsData[] | TableDiagnosisData[] | TableAnimalsData[];
}

export const renderCell = (
  col: Column,
  item: TableAccountsData | TableDiagnosisData | TableAnimalsData,
): React.ReactNode => {
  if (col.render) {
    return col.render(item);
  }

  return item[col.accessor as keyof typeof item];
};

export const SelectableCell: React.FC<SelectableCellProps> = ({
  isSelected,
  onSelect,
}) => (
  <TableCell>
    <Checkbox name="select" checked={isSelected} onChange={onSelect} />
  </TableCell>
);

export const ActionCell: React.FC<ActionCellProps> = ({ onActionClick }) => (
  <TableCell>
    <ActionButton onClick={onActionClick}>
      <MoreVertical width={17} height={17} />
    </ActionButton>
  </TableCell>
);

export const getIsSelected = (
  selectedRowsSet: Set<string> | string[],
  itemId: string,
): boolean => {
  if (selectedRowsSet instanceof Set) {
    return selectedRowsSet.has(itemId); // Usando has para Set
  }
  return (selectedRowsSet as string[]).includes(itemId); // Usando includes para Array
};

export const renderActionCell = ({
  action,
  selectMode,
  isSelected,
  toggleSelectRow,
  item,
  rowIndex,
  onAction,
}: RenderActionCellProps) => {
  if (!action) return null;

  return selectMode ? (
    <SelectableCell
      isSelected={isSelected}
      onSelect={() => toggleSelectRow?.(item.id)}
    />
  ) : (
    <ActionCell onActionClick={() => onAction(rowIndex)} />
  );
};

export const renderActionModal = ({
  activeRowIndex,
  rowIndex,
  action,
  modalRef,
  data,
}: RenderActionModalProps) => {
  if (activeRowIndex !== rowIndex) return null;

  return (
    <ActionModal ref={modalRef}>
      <ActionButtons
        action={action}
        onDelete={() => {}}
        activeRowIndex={activeRowIndex}
        data={data}
      />
    </ActionModal>
  );
};
