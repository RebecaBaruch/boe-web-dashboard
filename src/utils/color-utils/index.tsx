// function to get the color of the risk based on the value
export const getRiskColor = (risk: number): string => {
  if (risk >= 80) {
    return '#FF5454';
  } else if (risk >= 40) {
    return '#FFEB36';
  } else {
    return '#3BD640';
  }
};

export const getRiskTextColor = (risk: number): string => {
  if (risk >= 40 && risk < 80) {
    return '#474000';
  } else {
    return '#FFFFFF';
  }
};

// function to get the color of the result based on the value
export const getResultColor = (result: string): string => {
  switch (result) {
    case 'Berne':
      return '#4A65CF';
    case 'Dermatite':
      return '#00DAB6';
    case 'Dermatofitose':
      return '#FF12CB';
    case 'Dermatofilose':
      return '#8515E8';
    default:
      return '#ffffff';
  }
};

export const getResultTextColor = (result: string): string => {
  switch (result) {
    case 'Dermatite':
      return '#001F3C';
    case null:
      return '#717171';
    default:
      return '#ffffff';
  }
};

export const getStatusColor = (status: string): string => {
  switch (status) {
    case 'Em andamento':
      return '#FFAD0F';
    case 'Concluído':
      return '#45A6FF';
    default:
      return '#F2F2F5';
  }
};

export const getStatusTextColor = (status: string): string => {
  switch (status) {
    case 'Em andamento':
      return '#4C3200';
    case 'Concluído':
      return '#FFFFFF';
    default:
      return '#717171';
  }
};
