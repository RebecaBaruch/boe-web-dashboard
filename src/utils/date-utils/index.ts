// Returns the current day of the week in Portuguese
export function getCurrentDayOfWeek(): string {
  const daysOfWeek = [
    'Domingo',
    'Segunda-feira',
    'Terça-feira',
    'Quarta-feira',
    'Quinta-feira',
    'Sexta-feira',
    'Sábado',
  ];
  const today = new Date();
  return daysOfWeek[today.getDay()];
}

// Returns the current date in the format "31 Oct, 2024"
export function getCurrentDate(): string {
  const today = new Date();
  const day = today.getDate();
  const months = [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec',
  ];
  const month = months[today.getMonth()];
  const year = today.getFullYear();

  return `${day} ${month}, ${year}`;
}

// Returns the current time in the format "18h30"
export function getCurrentTime(): string {
  const now = new Date();
  const hours = now.getHours().toString().padStart(2, '0');
  const minutes = now.getMinutes().toString().padStart(2, '0');

  return `${hours}h${minutes}`;
}