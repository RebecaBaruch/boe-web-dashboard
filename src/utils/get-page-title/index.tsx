import { RoutesUrls } from "../enums/routes-url";

export const getPageTitle = (pathname: string): string => {
  const titles: Record<string, string> = {
    [RoutesUrls.BASE_URL]: 'Dashboard',
    [RoutesUrls.LINKED_ACCOUNTS]: 'Contas associadas',
    [RoutesUrls.DIAGNOSIS_HISTORIC]: 'Histórico de Análises',
    [RoutesUrls.USER_PROFILE]: 'Meu perfil',
    [RoutesUrls.LOGIN]: 'Login',
    [RoutesUrls.REGISTER]: 'Registro',
    [RoutesUrls.REGISTER_FARM]: 'Registro - Fazenda',
  };
  return titles[pathname] || '';
};