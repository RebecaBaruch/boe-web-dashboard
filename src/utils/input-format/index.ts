export const formatPhone = (value: string) => {
  const onlyNumbers = value.replace(/\D/g, '');

  const maxLength = 11;

  if (onlyNumbers.length == maxLength) {
    return `(${onlyNumbers.slice(0, 2)}) ${onlyNumbers.slice(2, 7)}-${onlyNumbers.slice(7, 11)}`;
  }
};

export const formatCep = (value: string) => {
  const onlyNumbers = value.replace(/\D/g, '');

  const maxLength = 8;

  if (onlyNumbers.length == maxLength) {
    return `${onlyNumbers.slice(0, 5)}-${onlyNumbers.slice(5, 8)}`;
  }
};

export const formatCnpj = (value: string) => {
  const onlyNumbers = value.replace(/\D/g, '');

  const maxLength = 14;

  if (onlyNumbers.length == maxLength) {
    return `${onlyNumbers.slice(0, 2)}.${onlyNumbers.slice(2, 5)}.${onlyNumbers.slice(5, 8)}/${onlyNumbers.slice(8, 12)}-${onlyNumbers.slice(12, 14)}`;
  }
};

export const togglePasswordVisibility = (
  isVisible: boolean,
  value: string,
): { inputType: string; displayValue: string } => {
  return {
    inputType: isVisible ? 'text' : 'password',
    displayValue: isVisible ? value : '*'.repeat(value.length), // Oculta a senha com asteriscos
  };
};

export const isValidEmail = (email: string | undefined) => {
  if (!email) return false;

  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
};
