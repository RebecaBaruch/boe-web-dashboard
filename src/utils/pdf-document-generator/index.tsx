import React from 'react';
import {
  Document,
  Page,
  Text,
  View,
  Image,
  StyleSheet,
} from '@react-pdf/renderer';
import { TableDiagnosisData } from 'components/data-table/types';
import {
  getResultColor,
  getResultTextColor,
  getStatusColor,
  getStatusTextColor,
} from 'utils/color-utils';

const styles = StyleSheet.create({
  page: {
    fontFamily: 'Helvetica',
    padding: 20,
  },
  header: {
    fontSize: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  sectionTitle: {
    fontSize: 14,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  animalPhoto: {
    width: 40,
    height: 40,
    borderRadius: 7,
  },
  animalInfoContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  analysisContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 20,
    borderBottom: '1 solid #00000034',
  },
  resultContainer: {
    flexDirection: 'row',
    gap: 20,
    marginBottom: 20,
  },
  resultBox: {
    flexDirection: 'column',
    gap: 5,
  },
  descriptionContainer: {
    flexDirection: 'row',
    gap: 20,
    borderRadius: 10,
    border: '1 solid #00000034',
  },
  descriptionBox: {
    flexDirection: 'column',
    gap: 5,
  },
  analysisImg: {
    display: 'flex',
    justifyContent: 'center',
    alignContent: 'center',
    marginRight: 32,
    width: 81,
    height: 81,
    borderRadius: 10,
    overflow: 'hidden',
  },
  badge: {
    padding: 4,
    borderRadius: 8,
    fontSize: 12,
    display: 'flex',
    marginBottom: 4,
  },
  authorContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: 50,
    padding: '4px 15px',
    borderRadius: 10,
    backgroundColor: '#f2f2f5',
  },
  authorPhoto: {
    width: 30,
    height: 30,
    borderRadius: '50%',
    marginRight: 8,
  },
  authorName: {
    fontSize: 12,
  },
});

interface PDFDocumentProps {
  record: TableDiagnosisData;
}

export const PDFDocumentGenerator: React.FC<PDFDocumentProps> = ({
  record,
}) => {
  const defaultAnimalImg = '../../../public/assets/animal-img.png';
  const defaultAnimalPic = '/public/assets/pic-image.png';

  const resultBadgeStyle = {
    backgroundColor: getResultColor(record.result),
    color: getResultTextColor(record.result),
  };

  const statusBadgeStyle = {
    backgroundColor: getStatusColor(record.status || 'Sem tratamento'),
    color: getStatusTextColor(record.status || 'Sem tratamento'),
  };

  return (
    <Document>
      <Page style={styles.page}>
        <View style={styles.header}>
          <Text>
            <b>Pré-diagnóstico</b>
          </Text>
          <Text>
            <b>Data de registro</b> {record.date}
          </Text>
        </View>

        <View style={styles.analysisContainer}>
          <View style={styles.animalInfoContainer}>
            <Image
              src={record.animalPhoto || defaultAnimalImg}
              style={styles.animalPhoto}
            />
            <View style={{ marginLeft: 10 }}>
              <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
                {record.animalId}
              </Text>
              <Text style={{ fontSize: 12 }}>{record.animalName}</Text>
            </View>
          </View>
          <View style={styles.authorContainer}>
            {record.authorPhoto ? (
              <Image src={record.authorPhoto} style={styles.authorPhoto} />
            ) : (
              <View style={styles.authorPhoto} />
            )}
            <Text style={styles.authorName}>{record.authorName}</Text>
          </View>
        </View>

        <View style={styles.resultContainer}>
          <View style={styles.resultBox}>
            <Text style={{ fontSize: 14, fontWeight: 'bold' }}>Resultado</Text>
            <View style={[styles.badge, resultBadgeStyle]}>
              <Text>{record.result}</Text>
            </View>
          </View>
          <View style={styles.resultBox}>
            <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
              Status de tratamento
            </Text>
            <View style={[styles.badge, statusBadgeStyle]}>
              <Text>{record.status || 'Sem tratamento'}</Text>
            </View>
          </View>
        </View>

        <View style={styles.analysisContainer}>
          <Image
            src={record.analysisImg || defaultAnimalPic}
            style={styles.analysisImg}
          />
          <View style={styles.descriptionBox}>
            <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
              Descrição da análise
            </Text>
            <Text style={{ fontSize: 10, fontWeight: 'light' }}>
              Probabilidade baixa, significa que as lesões ou sintomas
              observados não correspondem aos sinais típicos da doença.
            </Text>
          </View>
        </View>

        <Text style={{ fontSize: 10, color: '#666', marginTop: 30 }}>Böe</Text>
      </Page>
    </Document>
  );
};
