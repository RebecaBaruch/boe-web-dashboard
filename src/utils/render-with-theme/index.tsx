import React from 'react';
import { ThemeContextProvider } from '@telefonica/mistica';
import { render, RenderResult } from '@testing-library/react';
import { theme } from '../../config/theme';

export const renderWithTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeContextProvider theme={theme}>{children}</ThemeContextProvider>,
  );
};
