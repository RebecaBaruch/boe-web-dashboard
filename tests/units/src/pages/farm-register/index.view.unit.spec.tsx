import React from 'react';
import { screen } from '@testing-library/react';
import FarmRegister from '../../../../../src/pages/register/farm-register/view/farm-register.view';
import { renderWithTheme } from '../../../../../src/utils/render-with-theme';

describe('FarmRegister Component', () => {
  beforeEach(() => {
    renderWithTheme(<FarmRegister />);
  });

  it('should render the farm registration form with fields and buttons', () => {
    expect(screen.getByText('Nome da propriedade')).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText('Insira o nome da fazenda'),
    ).toBeInTheDocument();

    expect(screen.getByText('Telefone (fazenda)')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('(00) 00000-0000')).toBeInTheDocument();

    expect(screen.getByText('CNPJ')).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText('0.000.000/0000-00'),
    ).toBeInTheDocument();

    expect(screen.getByText('CEP')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('00000-000')).toBeInTheDocument();

    expect(screen.getByText('Estado')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('UF')).toBeInTheDocument();

    expect(screen.getByText('Cidade')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Localidade')).toBeInTheDocument();

    expect(screen.getByText('Endereço')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Bairro, Rua, nº')).toBeInTheDocument();

    expect(screen.getByText('Concluir registro')).toBeInTheDocument();
    expect(screen.getByText('Voltar')).toBeInTheDocument();
  });

  it('should render the farm data section header correctly', () => {
    expect(screen.getByText('Dados da fazenda')).toBeInTheDocument();
  });
});
