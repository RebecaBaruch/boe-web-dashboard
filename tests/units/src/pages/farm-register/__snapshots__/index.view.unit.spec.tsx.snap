// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FarmRegister should render FarmRegister with text FarmRegister 1`] = `
<div
  style="isolation: isolate;"
>
  <h1>
    FarmRegister
  </h1>
</div>
`;
